### ISSUE TITLE
Multiple "createAlias" don't work

### ISSUE THREAD

**Issue by dtamm** [fabiooshiro/plastic-criteria#16]
Given this example service method which will run into an exception:

```
def method() {
    def testdomain = Testdomain.createCriteria().list() {
        and {
            createAlias("domainB", "domainB") //Will work
            createAlias("domainB.domainC", "domainC") //Wont work with Exception "no such variable 'domainB.domainC' for class Testdomain"
            //Further filters for the query
        }
    }
}
```

Since i have multiple services which check for variables not only on the "surface" of an domain, this would be (not only for me i guess) a huge profit, since someone does not has to rewrite his service queries.

**Comment by dtamm** [comment]
I think that it had to do with what i wrote into the "mockCriteria"-method. After some more tests it works for me now.

**Comment by fabiooshiro** [comment]
What grails version you use?

**Comment by dtamm** [comment]
Using Grails 2.5.4, but for the moment, the error seems to be gone! But i will open another issue (countDistinct isnt working)

**Comment by dtamm** [comment]
I recently discovered the same problems again, and took a look into the code. I managed to find the problem, shall I do a pull request?

**Comment by fabiooshiro** [comment]
Yes plz

**Comment by dtamm** [comment]
Changes implemented in PR #19

**Comment by dtamm** [comment]
Also, did run the tests, no problems here :-)
